File: backend/app/schemas/caballo.py
from pydantic import BaseModel, Field, UUID4
from typing import Optional, Dict, Any, List
from datetime import date, datetime
from decimal import Decimal
from app.models.caballo import SexoCaballoEnum, EstadoCaballoEnum, ManejoEnum, TipoTrabajoEnum, CategoriaSanitariaEnum


# ========== CABALLO ==========

class CaballoBase(BaseModel):
    """Schema base de Caballo"""
    nombre: str = Field(..., min_length=1, max_length=100)
    raza: Optional[str] = Field(None, max_length=100)
    edad: Optional[int] = Field(None, ge=0, le=50)
    fecha_nacimiento: Optional[date] = None
    sexo: Optional[SexoCaballoEnum] = None
    color: Optional[str] = Field(None, max_length=50)
    altura: Optional[Decimal] = Field(None, ge=0, le=300)  # Altura en centímetros (máx 300cm = 3m)
    peso: Optional[Decimal] = Field(None, ge=0, le=2000)  # Peso en kilogramos (máx 2000kg)
    box_asignado: Optional[str] = Field(None, max_length=50)
    caracteristicas: Optional[str] = None


class CaballoCreate(CaballoBase):
    """Schema para crear Caballo"""
    # Campos OBLIGATORIOS
    numero_chip: str = Field(..., min_length=1, max_length=50)

    # Campos opcionales
    id_fomento: Optional[str] = Field(None, max_length=50)
    pedigree: Optional[str] = None
    propietario_id: Optional[UUID4] = None

    # Alimentación
    grano_balanceado: Optional[str] = Field(None, max_length=200)
    suplementos: Optional[str] = Field(None, max_length=500)
    cantidad_comidas_dia: Optional[int] = Field(None, ge=1, le=10)
    detalles_alimentacion: Optional[str] = None

    # Manejo
    tipo_manejo: Optional[ManejoEnum] = None

    # Trabajo
    dias_trabajo: Optional[str] = Field(None, max_length=100)
    dias_descanso: Optional[str] = Field(None, max_length=100)
    jinete_asignado: Optional[str] = Field(None, max_length=100)
    tiempo_trabajo_diario: Optional[int] = Field(None, ge=0, le=480)  # máx 8 horas
    trabajo_config: Optional[Dict[str, Any]] = None

    # Otros detalles
    embocadura_1: Optional[str] = Field(None, max_length=200)
    embocadura_2: Optional[str] = Field(None, max_length=200)
    cuidados_especiales: Optional[str] = None
    otra_info_1: Optional[str] = None
    otra_info_2: Optional[str] = None
    foto_principal: Optional[str] = None

    # Plan sanitario
    categoria_sanitaria: Optional[CategoriaSanitariaEnum] = None


class CaballoUpdate(BaseModel):
    """Schema para actualizar Caballo"""
    nombre: Optional[str] = Field(None, min_length=1, max_length=100)
    numero_chip: Optional[str] = Field(None, min_length=1, max_length=50)
    id_fomento: Optional[str] = Field(None, max_length=50)
    raza: Optional[str] = Field(None, max_length=100)
    edad: Optional[int] = Field(None, ge=0, le=50)
    fecha_nacimiento: Optional[date] = None
    pedigree: Optional[str] = None
    sexo: Optional[SexoCaballoEnum] = None
    color: Optional[str] = Field(None, max_length=50)
    altura: Optional[Decimal] = Field(None, ge=0, le=300)  # Altura en centímetros
    peso: Optional[Decimal] = Field(None, ge=0, le=2000)  # Peso en kilogramos
    propietario_id: Optional[UUID4] = None
    box_asignado: Optional[str] = Field(None, max_length=50)
    estado: Optional[EstadoCaballoEnum] = None
    caracteristicas: Optional[str] = None

    # Alimentación
    grano_balanceado: Optional[str] = Field(None, max_length=200)
    suplementos: Optional[str] = Field(None, max_length=500)
    cantidad_comidas_dia: Optional[int] = Field(None, ge=1, le=10)
    detalles_alimentacion: Optional[str] = None

    # Manejo
    tipo_manejo: Optional[ManejoEnum] = None

    # Trabajo
    dias_trabajo: Optional[str] = Field(None, max_length=100)
    dias_descanso: Optional[str] = Field(None, max_length=100)
    jinete_asignado: Optional[str] = Field(None, max_length=100)
    tiempo_trabajo_diario: Optional[int] = Field(None, ge=0, le=480)
    trabajo_config: Optional[Dict[str, Any]] = None

    # Otros detalles
    embocadura_1: Optional[str] = Field(None, max_length=200)
    embocadura_2: Optional[str] = Field(None, max_length=200)
    cuidados_especiales: Optional[str] = None
    otra_info_1: Optional[str] = None
    otra_info_2: Optional[str] = None
    foto_principal: Optional[str] = None

    # Plan sanitario
    categoria_sanitaria: Optional[CategoriaSanitariaEnum] = None


class CaballoSchema(CaballoBase):
    """Schema de respuesta de Caballo"""
    id: UUID4
    numero_chip: str
    id_fomento: Optional[str] = None
    pedigree: Optional[str] = None
    propietario_id: Optional[UUID4] = None
    estado: EstadoCaballoEnum
    qr_code: Optional[str] = None

    # Alimentación
    grano_balanceado: Optional[str] = None
    suplementos: Optional[str] = None
    cantidad_comidas_dia: Optional[int] = None
    detalles_alimentacion: Optional[str] = None

    # Manejo
    tipo_manejo: Optional[ManejoEnum] = None

    # Trabajo
    dias_trabajo: Optional[str] = None
    dias_descanso: Optional[str] = None
    jinete_asignado: Optional[str] = None
    tiempo_trabajo_diario: Optional[int] = None
    trabajo_config: Optional[Dict[str, Any]] = None

    # Otros detalles
    embocadura_1: Optional[str] = None
    embocadura_2: Optional[str] = None
    cuidados_especiales: Optional[str] = None
    otra_info_1: Optional[str] = None
    otra_info_2: Optional[str] = None

    # Plan sanitario
    categoria_sanitaria: Optional[CategoriaSanitariaEnum] = None
    foto_principal: Optional[str] = None

    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ========== FOTOS ==========

class FotoCaballoBase(BaseModel):
    """Schema base de Foto Caballo"""
    url: str
    descripcion: Optional[str] = Field(None, max_length=255)
    es_principal: bool = False


class FotoCaballoCreate(FotoCaballoBase):
    """Schema para crear Foto Caballo"""
    caballo_id: UUID4


class FotoCaballoUpdate(BaseModel):
    """Schema para actualizar Foto Caballo"""
    descripcion: Optional[str] = Field(None, max_length=255)
    es_principal: Optional[bool] = None


class FotoCaballoSchema(FotoCaballoBase):
    """Schema de respuesta de Foto Caballo"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime

    class Config:
        from_attributes = True


# ========== VACUNAS ==========

class VacunaRegistroBase(BaseModel):
    """Schema base de Vacuna"""
    tipo: str = Field(..., max_length=100)
    fecha: date
    veterinario: Optional[str] = Field(None, max_length=200)
    marca: Optional[str] = Field(None, max_length=200)
    frecuencia_dias: Optional[int] = Field(None, ge=1)
    proxima_fecha: Optional[date] = None
    observaciones: Optional[str] = None
    aplicada: bool = True


class VacunaRegistroCreate(VacunaRegistroBase):
    """Schema para crear Vacuna"""
    caballo_id: UUID4


class VacunaRegistroUpdate(BaseModel):
    """Schema para actualizar Vacuna"""
    tipo: Optional[str] = Field(None, max_length=100)
    fecha: Optional[date] = None
    veterinario: Optional[str] = Field(None, max_length=200)
    marca: Optional[str] = Field(None, max_length=200)
    frecuencia_dias: Optional[int] = Field(None, ge=1)
    proxima_fecha: Optional[date] = None
    observaciones: Optional[str] = None
    aplicada: Optional[bool] = None


class VacunaRegistroSchema(VacunaRegistroBase):
    """Schema de respuesta de Vacuna"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ========== REVISIÓN DENTAL ==========

class RevisionDentalBase(BaseModel):
    """Schema base de Revisión Dental"""
    fecha: date
    veterinario: Optional[str] = Field(None, max_length=200)
    observaciones: Optional[str] = None
    proxima_revision: Optional[date] = None


class RevisionDentalCreate(RevisionDentalBase):
    """Schema para crear Revisión Dental"""
    caballo_id: UUID4


class RevisionDentalUpdate(BaseModel):
    """Schema para actualizar Revisión Dental"""
    fecha: Optional[date] = None
    veterinario: Optional[str] = Field(None, max_length=200)
    observaciones: Optional[str] = None
    proxima_revision: Optional[date] = None


class RevisionDentalSchema(RevisionDentalBase):
    """Schema de respuesta de Revisión Dental"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime

    class Config:
        from_attributes = True


# ========== ESTUDIOS MÉDICOS ==========

class EstudioMedicoBase(BaseModel):
    """Schema base de Estudio Médico"""
    tipo: str = Field(..., max_length=100)
    fecha: date
    veterinario: Optional[str] = Field(None, max_length=200)
    zona_estudiada: Optional[str] = Field(None, max_length=200)
    diagnostico: Optional[str] = None
    archivo_url: Optional[str] = Field(None, max_length=500)
    observaciones: Optional[str] = None


class EstudioMedicoCreate(EstudioMedicoBase):
    """Schema para crear Estudio Médico"""
    caballo_id: UUID4


class EstudioMedicoUpdate(BaseModel):
    """Schema para actualizar Estudio Médico"""
    tipo: Optional[str] = Field(None, max_length=100)
    fecha: Optional[date] = None
    veterinario: Optional[str] = Field(None, max_length=200)
    zona_estudiada: Optional[str] = Field(None, max_length=200)
    diagnostico: Optional[str] = None
    archivo_url: Optional[str] = Field(None, max_length=500)
    observaciones: Optional[str] = None


class EstudioMedicoSchema(EstudioMedicoBase):
    """Schema de respuesta de Estudio Médico"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime

    class Config:
        from_attributes = True


# ========== HERRAJES ==========

class HerrajeRegistroBase(BaseModel):
    """Schema base de Herraje"""
    fecha: date
    herrador: Optional[str] = Field(None, max_length=200)
    observaciones: Optional[str] = None
    proximo_herraje: Optional[date] = None
    costo: Optional[Decimal] = Field(None, ge=0)


class HerrajeRegistroCreate(HerrajeRegistroBase):
    """Schema para crear Herraje"""
    caballo_id: UUID4


class HerrajeRegistroUpdate(BaseModel):
    """Schema para actualizar Herraje"""
    fecha: Optional[date] = None
    herrador: Optional[str] = Field(None, max_length=200)
    observaciones: Optional[str] = None
    proximo_herraje: Optional[date] = None
    costo: Optional[Decimal] = Field(None, ge=0)


class HerrajeRegistroSchema(HerrajeRegistroBase):
    """Schema de respuesta de Herraje"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime

    class Config:
        from_attributes = True


# ========== ANTIPARASITARIOS ==========

class AntiparasitarioRegistroBase(BaseModel):
    """Schema base de Antiparasitario"""
    fecha: date
    marca: Optional[str] = Field(None, max_length=200)
    drogas: Optional[str] = Field(None, max_length=500)
    dosis: Optional[str] = Field(None, max_length=100)
    proxima_aplicacion: Optional[date] = None
    observaciones: Optional[str] = None


class AntiparasitarioRegistroCreate(AntiparasitarioRegistroBase):
    """Schema para crear Antiparasitario"""
    caballo_id: UUID4


class AntiparasitarioRegistroUpdate(BaseModel):
    """Schema para actualizar Antiparasitario"""
    fecha: Optional[date] = None
    marca: Optional[str] = Field(None, max_length=200)
    drogas: Optional[str] = Field(None, max_length=500)
    dosis: Optional[str] = Field(None, max_length=100)
    proxima_aplicacion: Optional[date] = None
    observaciones: Optional[str] = None


class AntiparasitarioRegistroSchema(AntiparasitarioRegistroBase):
    """Schema de respuesta de Antiparasitario"""
    id: UUID4
    caballo_id: UUID4
    created_at: datetime

    class Config:
        from_attributes = True


# ========== ESQUEMA COMPLETO CON RELACIONES ==========

class CaballoCompleto(CaballoSchema):
    """Schema de Caballo con todas sus relaciones"""
    fotos: List[FotoCaballoSchema] = []
    vacunas: List[VacunaRegistroSchema] = []
    herrajes: List[HerrajeRegistroSchema] = []
    antiparasitarios: List[AntiparasitarioRegistroSchema] = []

    class Config:
        from_attributes = True


# ========== PLAN SANITARIO ==========

class ActividadPlanSanitario(BaseModel):
    """Actividad del plan sanitario"""
    tipo: str  # vacuna, desparasitacion, analisis
    nombre: str
    descripcion: str
    realizada: bool  # Si ya fue realizada este año
    fecha_realizada: Optional[date] = None  # Fecha en que se realizó
    proxima_fecha: Optional[date] = None  # Próxima vez que debe hacerse


class MesPlanSanitario(BaseModel):
    """Mes del plan sanitario"""
    mes: int
    mes_nombre: str
    actividades: List[ActividadPlanSanitario]


class PlanSanitarioResponse(BaseModel):
    """Respuesta con el plan sanitario del caballo"""
    categoria: Optional[str] = None  # A, B, o None si no tiene categoría
    nombre_categoria: Optional[str] = None
    descripcion: Optional[str] = None
    costo_mensual: Optional[int] = None  # Costo mensual del plan
    dosis_anuales: Optional[Dict[str, int]] = None  # Dosis anuales por tipo
    calendario: List[MesPlanSanitario] = []
    anio: int  # Año del plan


class ActividadPendiente(BaseModel):
    """Actividad pendiente del plan sanitario"""
    mes: int
    mes_nombre: str
    tipo: str
    nombre: str
    descripcion: str
    dias_vencido: Optional[int] = None  # Días desde que venció (positivo = vencido)


class EstadisticasPlanSanitario(BaseModel):
    """Estadísticas de cumplimiento del plan sanitario"""
    categoria: Optional[str] = None
    tiene_plan: bool
    costo_mensual: Optional[int] = None
    costo_anual: Optional[int] = None
    anio: int
    total_actividades: int
    actividades_realizadas: int
    actividades_pendientes: int
    porcentaje_cumplimiento: float
    proximas_actividades: List[ActividadPendiente]
    actividades_vencidas: List[ActividadPendiente]

File: frontend/src/types/caballo.ts
// ========== ENUMS ==========

export type SexoCaballo = 'macho' | 'hembra' | 'castrado';
export type EstadoCaballo = 'activo' | 'retirado' | 'en_tratamiento' | 'fallecido';
export type ManejoTipo = 'box' | 'box_piquete' | 'piquete' | 'palenque' | 'cross_tie';
export type CategoriaSanitaria = 'A' | 'B';

export interface TrabajoConfig {
  caminador?: boolean;
  cuerda?: boolean;
  manga?: boolean;
  montado?: boolean;
}

// ========== CABALLO ==========

export interface Caballo {
  id: string;

  // Datos básicos OBLIGATORIOS
  nombre: string;
  numero_chip: string;  // OBLIGATORIO - único

  // Datos básicos opcionales
  id_fomento?: string;
  edad?: number;
  fecha_nacimiento?: string;
  pedigree?: string;
  raza?: string;
  sexo?: SexoCaballo;
  color?: string;
  altura?: number;
  peso?: number;

  // Propietario y ubicación
  propietario_id?: string;
  propietario?: {
    id: string;
    nombre: string;
    apellido: string;
  };
  box_asignado?: string;
  estado: EstadoCaballo;

  // QR Code
  qr_code?: string;

  // Alimentación
  grano_balanceado?: string;
  suplementos?: string;
  cantidad_comidas_dia?: number;
  detalles_alimentacion?: string;

  // Manejo diario
  tipo_manejo?: ManejoTipo;

  // Trabajo diario
  dias_trabajo?: string;
  dias_descanso?: string;
  jinete_asignado?: string;
  tiempo_trabajo_diario?: number;  // en minutos
  trabajo_config?: TrabajoConfig;

  // Otros detalles
  embocadura_1?: string;
  embocadura_2?: string;
  cuidados_especiales?: string;
  caracteristicas?: string;
  otra_info_1?: string;
  otra_info_2?: string;
  foto_principal?: string;

  // Plan sanitario
  categoria_sanitaria?: CategoriaSanitaria;

  // Timestamps
  created_at: string;
  updated_at: string;
}

export interface CaballoCreate {
  // OBLIGATORIO
  nombre: string;
  numero_chip: string;

  // Opcional
  id_fomento?: string;
  edad?: number;
  fecha_nacimiento?: string;
  pedigree?: string;
  raza?: string;
  sexo?: SexoCaballo;
  color?: string;
  altura?: number;
  peso?: number;
  propietario_id?: string;
  box_asignado?: string;

  // Alimentación
  grano_balanceado?: string;
  suplementos?: string;
  cantidad_comidas_dia?: number;
  detalles_alimentacion?: string;

  // Manejo
  tipo_manejo?: ManejoTipo;

  // Trabajo
  dias_trabajo?: string;
  dias_descanso?: string;
  jinete_asignado?: string;
  tiempo_trabajo_diario?: number;
  trabajo_config?: TrabajoConfig;

  // Otros
  embocadura_1?: string;
  embocadura_2?: string;
  cuidados_especiales?: string;
  caracteristicas?: string;
  otra_info_1?: string;
  otra_info_2?: string;
  foto_principal?: string;

  // Plan sanitario
  categoria_sanitaria?: CategoriaSanitaria;
}

export interface CaballoUpdate {
  nombre?: string;
  numero_chip?: string;
  id_fomento?: string;
  edad?: number;
  fecha_nacimiento?: string;
  pedigree?: string;
  raza?: string;
  sexo?: SexoCaballo;
  color?: string;
  altura?: number;
  peso?: number;
  propietario_id?: string;
  box_asignado?: string;
  estado?: EstadoCaballo;

  // Alimentación
  grano_balanceado?: string;
  suplementos?: string;
  cantidad_comidas_dia?: number;
  detalles_alimentacion?: string;

  // Manejo
  tipo_manejo?: ManejoTipo;

  // Trabajo
  dias_trabajo?: string;
  dias_descanso?: string;
  jinete_asignado?: string;
  tiempo_trabajo_diario?: number;
  trabajo_config?: TrabajoConfig;

  // Otros
  embocadura_1?: string;
  embocadura_2?: string;
  cuidados_especiales?: string;
  caracteristicas?: string;
  otra_info_1?: string;
  otra_info_2?: string;
  foto_principal?: string;

  // Plan sanitario
  categoria_sanitaria?: CategoriaSanitaria;
}

// ========== FOTOS ==========

export interface FotoCaballo {
  id: string;
  caballo_id: string;
  url: string;
  descripcion?: string;
  es_principal: boolean;
  created_at: string;
}

export interface FotoCaballoCreate {
  caballo_id: string;
  url: string;
  descripcion?: string;
  es_principal?: boolean;
}

export interface FotoCaballoUpdate {
  descripcion?: string;
  es_principal?: boolean;
}

// ========== VACUNAS ==========

export interface VacunaRegistro {
  id: string;
  caballo_id: string;
  tipo: string;  // anemia, influenza, encefalomielitis, rinoneumonitis, rabia, perfil, etc.
  fecha: string;
  veterinario?: string;
  marca?: string;
  frecuencia_dias?: number;
  proxima_fecha?: string;  // Calculado automáticamente
  observaciones?: string;
  aplicada: boolean;
  created_at: string;
  updated_at: string;
}

export interface VacunaRegistroCreate {
  caballo_id: string;
  tipo: string;
  fecha: string;
  veterinario?: string;
  marca?: string;
  frecuencia_dias?: number;
  proxima_fecha?: string;
  observaciones?: string;
  aplicada?: boolean;
}

export interface VacunaRegistroUpdate {
  tipo?: string;
  fecha?: string;
  veterinario?: string;
  marca?: string;
  frecuencia_dias?: number;
  proxima_fecha?: string;
  observaciones?: string;
  aplicada?: boolean;
}

// ========== REVISIÓN DENTAL ==========

export interface RevisionDental {
  id: string;
  caballo_id: string;
  fecha: string;
  veterinario?: string;
  observaciones?: string;
  proxima_revision?: string;
  created_at: string;
}

export interface RevisionDentalCreate {
  caballo_id: string;
  fecha: string;
  veterinario?: string;
  observaciones?: string;
  proxima_revision?: string;
}

export interface RevisionDentalUpdate {
  fecha?: string;
  veterinario?: string;
  observaciones?: string;
  proxima_revision?: string;
}

// ========== ESTUDIO MÉDICO ==========

export interface EstudioMedico {
  id: string;
  caballo_id: string;
  tipo: string;  // radiografia, ecografia, resonancia, tomografia, endoscopia, otros
  fecha: string;
  veterinario?: string;
  zona_estudiada?: string;
  diagnostico?: string;
  archivo_url?: string;
  observaciones?: string;
  created_at: string;
}

export interface EstudioMedicoCreate {
  caballo_id: string;
  tipo: string;
  fecha: string;
  veterinario?: string;
  zona_estudiada?: string;
  diagnostico?: string;
  archivo_url?: string;
  observaciones?: string;
}

export interface EstudioMedicoUpdate {
  tipo?: string;
  fecha?: string;
  veterinario?: string;
  zona_estudiada?: string;
  diagnostico?: string;
  archivo_url?: string;
  observaciones?: string;
}

// ========== HERRAJE ==========

export interface HerrajeRegistro {
  id: string;
  caballo_id: string;
  fecha: string;
  herrador?: string;
  observaciones?: string;
  proximo_herraje?: string;
  costo?: number;
  created_at: string;
}

export interface HerrajeRegistroCreate {
  caballo_id: string;
  fecha: string;
  herrador?: string;
  observaciones?: string;
  proximo_herraje?: string;
  costo?: number;
}

export interface HerrajeRegistroUpdate {
  fecha?: string;
  herrador?: string;
  observaciones?: string;
  proximo_herraje?: string;
  costo?: number;
}

// ========== ANTIPARASITARIO ==========

export interface AntiparasitarioRegistro {
  id: string;
  caballo_id: string;
  fecha: string;
  marca?: string;
  drogas?: string;
  dosis?: string;
  proxima_aplicacion?: string;
  observaciones?: string;
  created_at: string;
}

export interface AntiparasitarioRegistroCreate {
  caballo_id: string;
  fecha: string;
  marca?: string;
  drogas?: string;
  dosis?: string;
  proxima_aplicacion?: string;
  observaciones?: string;
}

export interface AntiparasitarioRegistroUpdate {
  fecha?: string;
  marca?: string;
  drogas?: string;
  dosis?: string;
  proxima_aplicacion?: string;
  observaciones?: string;
}

// ========== CABALLO COMPLETO ==========

export interface CaballoCompleto extends Caballo {
  fotos: FotoCaballo[];
  vacunas: VacunaRegistro[];
  herrajes: HerrajeRegistro[];
  antiparasitarios: AntiparasitarioRegistro[];
}

// ========== HISTORIAL MÉDICO (legacy - mantener por compatibilidad) ==========

export interface HistorialMedico {
  id: string;
  caballo_id: string;
  tipo: 'vacuna' | 'desparasitacion' | 'tratamiento' | 'cirugia' | 'revision' | 'otro';
  fecha: string;
  veterinario?: string;
  descripcion: string;
  medicamento?: string;
  dosis?: string;
  proxima_aplicacion?: string;
  costo?: number;
  documentos?: string[];
  created_at: string;
  updated_at: string;
}

export interface PlanAlimentacion {
  id: string;
  caballo_id: string;
  tipo_alimento: string;
  cantidad_diaria: string;
  horarios: string[];
  suplementos?: string;
  restricciones?: string;
  updated_by?: string;
  created_at: string;
  updated_at: string;
}

// ========== PLAN SANITARIO 2026 ==========

export interface ActividadPlanSanitario {
  tipo: string; // vacuna, desparasitacion, analisis
  nombre: string;
  descripcion: string;
  realizada: boolean;
  fecha_realizada?: string;
  proxima_fecha?: string;
}

export interface MesPlanSanitario {
  mes: number;
  mes_nombre: string;
  actividades: ActividadPlanSanitario[];
}

export interface PlanSanitarioResponse {
  categoria?: string | null; // 'A', 'B', o null
  nombre_categoria?: string | null;
  descripcion?: string | null;
  costo_mensual?: number | null;
  dosis_anuales?: Record<string, number> | null;
  calendario: MesPlanSanitario[];
  anio: number;
}

export interface ActividadPendiente {
  mes: number;
  mes_nombre: string;
  tipo: string;
  nombre: string;
  descripcion: string;
  dias_vencido?: number | null;
}

export interface EstadisticasPlanSanitario {
  categoria?: string | null;
  tiene_plan: boolean;
  costo_mensual?: number | null;
  costo_anual?: number | null;
  anio: number;
  total_actividades: number;
  actividades_realizadas: number;
  actividades_pendientes: number;
  porcentaje_cumplimiento: number;
  proximas_actividades: ActividadPendiente[];
  actividades_vencidas: ActividadPendiente[];
}
